### PR TITLE
Fix/issue-3229-3230 [Admin panel][Reports] Decimal without zero before the dot is not accepted by the 'Сума (UAH)' field

### PR DESCRIPTION
**Pull request by yaroslavkucher**
## Github ticket

* [3229](https://github.com/ita-social-projects/VictoryCenter-Back/issues/3229)
* [3230](https://github.com/ita-social-projects/VictoryCenter-Back/issues/3230)

## Description

The main goal of this PR is to improve the user experience when entering monetary amounts. It automatically formats inputs that start with a decimal separator (a dot or a comma) by prepending a leading zero, preventing unnecessary validation errors for valid shorthand inputs like `.12` or `,50`.

## Summary of change

* Updated the `normalizeFundsExpendituresAmountInput` function to automatically prepend `0` if the value starts with `,` (or `.` normalized to `,`).
* Added support for negative shorthand values (e.g., prepending `-0,` if the input starts with `-,`).
* Added 4 new unit test cases in `funds-expenditures-record-schema.test.ts` to verify the new normalization logic for positive and negative shorthand inputs.
* Added 1 new unit test case to ensure that shorthand decimal inputs successfully pass the `validateFundsExpendituresAmount` check.

## How to recreate changes

1. Navigate to the page with the Add/Edit Funds Expenditures Record Modal.
2. Open the modal and focus on the Amount (UAH or USD) input field.
3. Type a shorthand decimal value, for example: `.12`, `,12`, or `-.12`.
4. Verify that the input is automatically formatted to `0,12` or `-0,12` respectively.
5. Confirm that no validation errors (such as "Amount must be a number") are triggered for these specific inputs upon change or blur.

## CHECK LIST
- [x] New tests was added or existing was modified
- [ ] Changes has severe impact on users
- [ ] Changes has medium impact on users
- [x] Changes has light impact on users
- [x] Test covetage was updated
- [x] PR meets all conventions

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved handling of decimal amount inputs without a leading zero, such as `.12` and `-.12`.
  * Decimal values using either commas or periods are now normalized consistently during validation.

* **Tests**
  * Added coverage for positive and negative decimal formats and validation of leading-decimal amounts.

### CODE REVIEW

File: src/validation/admin/reports-schema/funds-expenditures-record-schema/funds-expenditures-record-schema.ts [hunk 1]
```diff
@@ -19,7 +19,14 @@ export const normalizeFundsExpendituresAmountInput = (value: string | undefined
...
     const withNormalizedSpaces = value.replaceAll(/\s+/g, ' ').trimStart();
-    const withCommaSeparator = withNormalizedSpaces.replaceAll('.', ',');
+    let withCommaSeparator = withNormalizedSpaces.replaceAll('.', ',');
+
+    if (withCommaSeparator.startsWith(',')) {
+        withCommaSeparator = `0${withCommaSeparator}`;
+    } else if (withCommaSeparator.startsWith('-,')) {
+        withCommaSeparator = `-0,${withCommaSeparator.slice(2)}`;
+    }
+
```
Comment (mehalyna) [inline src/validation/admin/reports-schema/funds-expenditures-record-schema/funds-expenditures-record-schema.ts]: The logic for prepending zero to negative values with leading comma is incomplete and will produce incorrect results. Extract the sign separately and reapply it after normalization to handle all edge cases correctly.
```
const isNegative = withCommaSeparator.startsWith('-');
const absoluteValue = isNegative ? withCommaSeparator.slice(1) : withCommaSeparator;
const normalized = absoluteValue.startsWith(',') ? `0${absoluteValue}` : absoluteValue;
withCommaSeparator = isNegative ? `-${normalized}` : normalized;
```

File: src/validation/admin/reports-schema/funds-expenditures-record-schema/funds-expenditures-record-schema.test.ts [hunk 2]
```diff
@@ -24,6 +24,22 @@ describe('FUNDS_EXPENDITURES_RECORD_VALIDATION_FUNCTIONS', () => {
...
+            expect(normalizeFundsExpendituresAmountInput(',12')).toBe('0,12');
+        });
+
+        it('should prepend zero for negative values starting with a dot', () => {
+            expect(normalizeFundsExpendituresAmountInput('-.12')).toBe('-0,12');
+        });
+
+        it('should prepend zero for negative values starting with a comma', () => {
+            expect(normalizeFundsExpendituresAmountInput('-,12')).toBe('-0,12');
+        });
```
Comment (mehalyna) [inline src/validation/admin/reports-schema/funds-expenditures-record-schema/funds-expenditures-record-schema.test.ts:42]: The test cases only verify the normalization function with specific inputs but don't test the full integration through the validation function with edge cases like leading zeros with spaces or special characters. Add integration tests covering real user input scenarios to ensure end-to-end correctness.

```
it('should handle dot with spaces before decimal digits', () => {
    expect(normalizeFundsExpendituresAmountInput('. 12')).toBe('0, 12');
});
it('should validate negative comma-starting input in validation function', () => {
    expect(validateFundsExpendituresAmount('-,12', 'change')).toBeUndefined();
});
```

File: src/validation/admin/reports-schema/funds-expenditures-record-schema/funds-expenditures-record-schema.ts (hunk 1 above)
Comment (mehalyna) [inline src/validation/admin/reports-schema/funds-expenditures-record-schema/funds-expenditures-record-schema.ts:31]: The logic doesn't account for the case where a negative sign is followed immediately by a dot (e.g., "-.5"), which after comma normalization becomes "-,5" but the current code checks for "-," as a substring, failing to handle "- ." with spaces. Normalize spaces before checking for the negative-comma pattern.

**Review by Sofiya397 — APPROVED** [review]

**Review by mehalyna — APPROVED** [review]

**Review by cirillad — APPROVED** [review]
